### PR TITLE
gh-125832: Clarify comment for inlined comprehensions as per PEP-709

### PR DESCRIPTION
**Pull request by rimi0108**
* Issue: gh-125832

### CODE REVIEW

**Comment by rimi0108** [comment]
@corona10

**Comment by corona10** [comment]
@carljm @willingc Would you like to take a look? 
Carl, FYI, Hyerime Choi is the participant in this year's PyCon KR Sprint :)

File: Python/codegen.c [hunk 1]
```diff
@@ -4087,9 +4087,11 @@ codegen_call_helper(compiler *c, location loc,
...
 }
 
-/* List and set comprehensions and generator expressions work by creating a
-  nested function to perform the actual iteration. This means that the
-  iteration variables don't leak into the current scope.
+/* List and set comprehensions work by being inlined at the location where
+  they are defined. The isolation of iteration variables is provided by
+  pushing/popping clashing locals on the stack. Generator expressions work
+  by creating a nested function to perform the actual iteration.
+  This means that the iteration variables don't leak into the current scope.
```
Comment (rruuaanng) [inline Python/codegen.c]:
```suggestion
/* List and set comprehensions work by being inlined at the location where
   they are defined. The isolation of iteration variables is provided by
   pushing/popping clashing locals on the stack. Generator expressions work
   by creating a nested function to perform the actual iteration.
   This means that the iteration variables don't leak into the current scope.
```

File: Python/codegen.c [hunk 2]
```diff
@@ -4087,9 +4087,11 @@ codegen_call_helper(compiler *c, location loc,
...
 }
 
-/* List and set comprehensions and generator expressions work by creating a
-  nested function to perform the actual iteration. This means that the
-  iteration variables don't leak into the current scope.
+/* List and set comprehensions work by being inlined at the location where
+   they are defined. The isolation of iteration variables is provided by
+   pushing/popping clashing locals on the stack. Generator expressions work
+   by creating a nested function to perform the actual iteration.
+   This means that the iteration variables don't leak into the current scope.
```
Comment (willingc) [inline Python/codegen.c:4094]:
```suggestion
   This means that the iteration variables don't leak into the current scope.
   See https://peps.python.org/pep-0709/ for additional information.
```

**Review by willingc — APPROVED** [review]
Looks good! I have one suggestion to add a reference to PEP 709

**Comment by willingc** [comment]
Congrats @rimi0108 on the contribution during the sprint. :tada: cc/ @corona10 nice mentoring.

**Comment by picnixz** [comment]
I think this one can be backported to 3.12 since PEP-709 was for 3.12, right?

**Comment by willingc** [comment]
@picnixz Yep. 3.12 and 3.13. Thanks!

**Comment by carljm** [comment]
Sorry, I wasn't available over the weekend, but this change looks good to me! Thank you @rimi0108 for the PR and @willingc for reviewing :)

**Pull request merged**